src/twisted/python/_setup.py

```diff
@@ -81,34 +75,25 @@
     'twisted-dev-tools >= 0.0.2',
     'python-subunit',
     'sphinx >= 1.3.1',
-    'towncrier >= 17.4.0'
+    'towncrier >= 17.4.0',
+    'twistedchecker >= 0.7.2',
 ]
 
-if not _PY3:
-    # These modules do not yet work on Python 3.
-    _dev += [
-        'twistedchecker >= 0.4.0',
-        'pydoctor >= 16.2.0',
-    ]
-
 _EXTRA_OPTIONS = dict(
     dev=_dev,
     tls=[
         'pyopenssl >= 16.0.0',
         # service_identity 18.1.0 added support for validating IP addresses in
         # certificate subjectAltNames
         'service_identity >= 18.1.0',
-        # idna 2.3 introduced some changes that break a few things.  Avoid it.
-        # The problems were fixed in 2.4.
-        'idna >= 0.6, != 2.3',
+        'idna >= 2.4',
     ],
     conch=[
         'pyasn1',
         'cryptography >= 2.6',
         'appdirs >= 1.4.0',
         'bcrypt >= 3.0.0',
     ],
-    soap=['soappy'],
```

Comment:
lol `twisted.web.soap` still exists

Reply:
(Given its apparent total lack of test coverage I'm suggesting we delete it; anyone who wanted this functionality on python 3 would have noticed it was missing some time ago.)